### DISCUSSION TITLE
Specify compile location

### DISCUSSION THREAD

**Discussion by irsya0** [mfussenegger/nvim-jdtls#443] [Q&A]
I want to specify the location where the project is compiled. When compiling 2 new folders are created, one named like the project with some random letters and numbers after a underscore before it and one named jdtls-project. I want the project to be compiled in my current project location in ./out/.
How to specify the compile location?

**Comment by lobophf** [comment]
How do you usually compile your projects? Do you use any shortcuts in `nvim`? Do you use any Dependency Management like `Gradle` or `Maven`? It's possible that your question is not related to `nvim-jdtls`.

Remember the IDE itself is just a tool to help write code and the compiler compiles it.

You can create your project and use the command below to compile as you wish.
```bash
javac MyProject.java -d out/ 
``` 
To run it use the next command.
```bash
java -cp out/ MyProject
```

**Reply by irsya0** [reply]
I am using the builtin compile function of nvim-jdtls. I am not using any dependency managment system.

**Comment by lobophf** [comment]
I've never tried to compile anything using the `:lua require('jdtls').compile` command of nvim-jdtls. I believe it is not possible to pass arguments to the compiler that way, only the `full` and `incremental` options are allowed.

However, you can customize any command to behave the way you want. You can try using this shortcut to be executed when you hit the `c` key twice.

```lua
vim.api.nvim_set_keymap('n', 'cc', '<cmd>:!javac *.java -d out/ <CR>', {})
```
EDIT: Also, you can use [this](https://github.com/MunifTanjim/exrc.nvim) plugin to customize nvim commands locally for your projects.

**Reply by irsya0** [reply]
I tried your solution by running javac. This works when having everything in one package, but I am not able to compile bigger programs this way.

**Comment by lobophf** [comment]
That's a good reason to add `Gradle` or `Maven` to your project. You can find some weird alternative, but it will be impractical at some point.
